diffkemp/simpll/DifferentialGlobalNumberState.cpp

```diff
@@ -20,24 +20,74 @@
 uint64_t DifferentialGlobalNumberState::getNumber(GlobalValue *value) {
     auto number = GlobalNumbers.find(value);
     u_int64_t result;
+
     if (number == GlobalNumbers.end()) {
         // Get new number for the global value
         result = nextNumber;
-        GlobalNumbers.insert({value, nextNumber});
 
         // Try to find global value with the same name in the other module, if
         // it exists, assign it the same number
         auto otherModule = value->getParent() == First ? Second : First;
-        GlobalValue *otherValue = otherModule->getNamedValue(
-                value->getName());
-        if (otherValue)
-            GlobalNumbers.insert({otherValue, nextNumber});
+        auto GA = dyn_cast<GlobalVariable>(value);
+
+        // If the value is a GlobalVariable, has the attribute unnamed_addr and
+        // the initializer is either a ConstantDataSequential or a ConstantInt,
+        // use a map for comparison by value instead of name
+        if (GA && GA->hasGlobalUnnamedAddr() && GA->hasInitializer() &&
+            (dyn_cast<ConstantDataSequential>(GA->getInitializer()) ||
+             dyn_cast<ConstantInt>(GA->getInitializer()))) {
+            // Value is a constant that can be compared by value
+            Constant *I = GA->getInitializer();
+
+            if (ConstantDataSequential *CDS =
+                dyn_cast<ConstantDataSequential>(I)) {
+                // Value is a string - look it up in the string map
+                auto possibleResult = Strings.find(
+                    CDS->getAsString());
+
+                if(possibleResult != Strings.end()) {
+                    // If the string is in the map, set the result to it
+                    result = (*possibleResult).getValue();
```

Comment:
Use -> instead of (*).
Also, shouldn't we insert {value, result} into the GlobalNumbers map in this case?